src/NodeCompiler/CompileNodeToValue.php

```diff
@@ -140,15 +140,15 @@ private function compileClassConstFetch(Node\Expr\ClassConstFetch $node, Compile
         $classInfo = null;
 
         if ($className === 'self' || $className === 'static') {
-            $classInfo = $this->getConstantDeclaringClass($node->name, $context->getSelf());
+            $classInfo = $this->getConstantDeclaringClass((string) $node->name, $context->getSelf());
```

Comment:
This cast is unnecessary as `$node->name` will always be a string here. Maybe an `assert(is_string($node->name))` above would be better, and would make the other cast below unnecessary